shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,153 @@
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+@ApplicationScoped
+@ControllerConfiguration(name = ManagedProcessorController.NAME, labelSelector = LabelsBuilder.V2_RECONCILER_LABEL_SELECTOR)
+public class ManagedProcessorController implements Reconciler<ManagedProcessor>,
+        EventSourceInitializer<ManagedProcessor>, ErrorStatusHandler<ManagedProcessor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorController.class);
+    public static final String NAME = "managedprocessorcontroller";
+
+    @ConfigProperty(name = "event-bridge.executor.deployment.timeout-seconds")
+    int executorTimeoutSeconds;
+
+    @ConfigProperty(name = "event-bridge.executor.poll-interval.milliseconds")
+    int executorPollIntervalMilliseconds;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Inject
+    NetworkingService networkingService;
+
+    @Inject
+    OperatorMetricsService metricsService;
+
+    @Inject
+    BridgeErrorHelper bridgeErrorHelper;
+
+    @Override
+    public Map<String, EventSource> prepareEventSources(EventSourceContext<ManagedProcessor> eventSourceContext) {
+        return EventSourceInitializer.nameEventSources(
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V2_OPERATOR_NAME, ManagedProcessor.COMPONENT_NAME, Secret.class),
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V2_OPERATOR_NAME, ManagedProcessor.COMPONENT_NAME, ConfigMap.class),
+                EventSourceFactory.buildInformerFromPrimaryResource(eventSourceContext, LabelsBuilder.V2_OPERATOR_NAME, ManagedProcessor.COMPONENT_NAME, CamelIntegration.class),
+                networkingService.buildInformerEventSource(eventSourceContext, LabelsBuilder.V2_OPERATOR_NAME, ManagedProcessor.COMPONENT_NAME));
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        String managedProcessorName = managedProcessor.getMetadata().getName();
+        String managedProcessorNamespace = managedProcessor.getMetadata().getNamespace();
+
+        LOGGER.info("Reconciling ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        ManagedProcessorStatus processorStatus = managedProcessor.getStatus();
+
+        if (!processorStatus.isReadyV2() && isTimedOut(processorStatus)) {
+            // The only resource that can be in timeout state is Camel so let's use this to invalidate the ManagedProcessor
```

Comment:
Shouldn't we also log something here? Maybe even with warn level as there is a timout.

Reply:
Timeout is going to be removed see https://issues.redhat.com/browse/MGDOBR-1308